services/viva-ms/src/lambdas/submitCompletion.ts

```diff
@@ -0,0 +1,161 @@
+import { VIVA_COMPLETION_RECEIVED, VIVA_RANDOM_CHECK_RECEIVED } from '../libs/constants';
+import { getItem as getStoredUserCase } from '../libs/queries';
+import * as dynamoDb from '../libs/dynamoDb';
+import params from '../libs/params';
+import config from '../libs/config';
+import log from '../libs/logs';
+
+import caseHelper from '../helpers/createCase';
+import attachment from '../helpers/attachment';
+import vivaAdapter from '../helpers/vivaAdapterRequestClient';
+import { CaseAttachment } from '../helpers/attachment';
+
+import type { CaseItem, CaseForm } from '../types/caseItem';
+import type { EventDetailCaseKeys } from '../types/eventDetail';
+
+interface LambdaDetail {
+  readonly caseKeys: EventDetailCaseKeys;
+  readonly status?: Record<string, unknown>;
+  readonly state?: string;
+}
+
+export interface LambdaRequest {
+  readonly detail: LambdaDetail;
+}
+
+interface PostCompletionResponse {
+  readonly status: string;
+}
+
+interface PostCompletionRequest {
+  personalNumber: string;
+  workflowId: string | null | undefined;
+  attachments: CaseAttachment[];
+}
+
+interface GetStoredUserCaseResponse {
+  readonly Item: CaseItem | null;
+}
+
+interface SSMParamsReadResponse {
+  readonly randomCheckFormId: string;
+  readonly completionFormId: string;
+}
+
+export interface UpdateCaseResponse {
+  readonly Attributes: unknown;
+}
+
+interface UpdateCaseParameters {
+  caseKeys: EventDetailCaseKeys;
+  newState: typeof VIVA_COMPLETION_RECEIVED | typeof VIVA_RANDOM_CHECK_RECEIVED;
+  currentFormId: string;
+  initialCompletionForm: Record<string, CaseForm>;
+}
+
+export interface Dependencies {
+  getStoredUserCase: (
+    TableName: string,
+    PK: string,
+    SK: string
+  ) => Promise<[Error | null, GetStoredUserCaseResponse]>;
+  readParams: (name: string) => Promise<SSMParamsReadResponse>;
+  postCompletion: (payload: PostCompletionRequest) => Promise<PostCompletionResponse>;
+  updateCase: (params: UpdateCaseParameters) => Promise<UpdateCaseResponse | null>;
```

Comment:
The contents of `UpdateCaseResponse` doesn't seem to be used. Could probably replace it flat out with `unknown`:

```typescript
updateCase: (params: UpdateCaseParameters) => Promise<unknown>;
```